File: 1732_Find_the_Highest_Altitude/solution.py
# class Solution:
#     def largestAltitude(self, gain: List[int]) -> int:
#         n = len(gain)
#         result = [0] * (n + 1)

#         for i in range(n):
#             result[i + 1] = result[i] + gain[i]

#         return max(result)

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        temp, maxx = 0, 0
        for i in range(len(gain)):
            temp += gain[i]
            if temp > maxx:
                maxx = temp
        return maxx

File: 1732_Find_the_Highest_Altitude/solution.c
// int largestAltitude(int* gain, int gainSize) {
//     int* result = calloc((gainSize + 1), sizeof(int));
//     for (int i = 0; i < gainSize; i++) result[i + 1] = result[i] + gain[i];
//     int max = result[0];
//     for (int i = 0; i < gainSize + 1; i++) if (result[i] > max) max = result[i];
//     return max;
// }

int largestAltitude(int* gain, int gainSize) {
    int temp = 0, max = 0;
    for (int i = 0; i < gainSize; i++) {
        temp += gain[i];
        if (temp > max) max = temp;
    }
    return max;
}